### PR TITLE
1521 can we change elcid.models.appointment type to use choices

### LINKED ISSUE #1521: can we change elcid.models.appointment_type to use choices?
elcid.models.appointment_type could use the choices api. This will make the form more supportable.

- [x] change appointment_type to use the choices api
- [x] update the form to account for this
- [x] make and run migrations

### PR DESCRIPTION
**Pull request by pacharanero**
closes #1521 

image

### CODE REVIEW

**Comment by fredkingham** [comment]
@pacharanero can we have choices being both the same value. Its a style and consistency choice more than anything.

We don't use django choices in quite the same way as django as we only ever use the last field. So to make sure developers understand what's stored in the db/shown on the front end, we always use the first and last field in the individual choices field with the same values.

David wrote a helper function to help, because who likes typing?
http://opal.openhealthcare.org.uk/docs/reference/core_fields/

**Comment by pacharanero** [comment]
The helper function is neat, although much more 'magical' than just writing both bits out longhand.
A few weeks ago I wouldn't have known what was going on had I read

```python
    APPOINTMENT_CHOICES = enum(
        "Doctor Clinic review appointment",
        "Nurse Clinic review appointment")
```

 in the code. Whereas:

```python
    APPOINTMENT_CHOICES = [
        ("Doctor Clinic review appointment", "Doctor Clinic review appointment"),
        ("Nurse Clinic review appointment", "Nurse Clinic review appointment")
    ]
```
is just plain old Django, you can google it and work out what's going on.

**Comment by davidmiller** [comment]
You don't think you would have found `from opal.core.fields import enum` and then googled `openhealthcare opal enum`  ?

File: elcid/migrations/0059_auto_20180327_0922.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (davidmiller) [inline elcid/migrations/0059_auto_20180327_0922.py]: Can we take the field definition from this migration and put it directly into the previous one ?

I don't think there's much value in preserving the interim migration step ?

Comment (pacharanero) [reply elcid/migrations/0059_auto_20180327_0922.py]: agreed - wasn't sure how to fix but have done so following your instructions

**Comment by pacharanero** [comment]
> You don't think you would have found from opal.core.fields import enum and then googled openhealthcare opal enum ?

Maybe, maybe not. Obvious things are only obvious with a certain amount of context. Many new Opal devs will not yet have this context. I'm in between. 

For the difference it makes to the readability of the code, there's probably not a great advantage to using `enum` IMO. If it was a larger list we'd be using a `FreeTextOrForeignKeyField` and `LookupList` anyway.

**Comment by fredkingham** [comment]
👍

**Pull request merged**